### ISSUE TITLE
Is there a way to pull <style jsx> out of the return?

### ISSUE THREAD

**Issue by corysimmons** [zeit/styled-jsx#180]
Requires a lot of superfluous wrapping <divs> just for styling purposes.

I'm sure there's rationale behind it and it's not a deal-breaker, but can we start a discussion if it seems possible?

**Comment by giuseppeg** [comment]
It is not clear to me what you mean by _"pull <style jsx> out of the return"_

**Comment by tz5514** [comment]
@giuseppeg  I think he mean something like this: 
```jsx
const styles = (
  <style jsx>{`
    div {
      padding: 10px;
    }

    p {
      font-size: 20px;
    }
  `}</style>
)

export default () => (
  <div>
    <p>Hello</p>
    {styles}
  </div>
)
```
I'm looking for solution for this problem too.
Can we do this?

**Comment by corysimmons** [comment]
Sorry, I meant, does anyone have any ideas on how to inject styles?

As an example, if I have a stateless component that is just an `<h1>` tag, I have to wrap it in a superfluous `<div>` just to be able to use `<style jsx>` (the no adjacent rule).

It might be cool to do something like:

```jsx
const styles = `
  background: red;
  color: white;
`

export default () => <h1 styled-jsx={styles}>Foo</h1>
```

...

Nevermind... I think that would work...

😅 

https://www.youtube.com/watch?v=bf9d7rSf_Ks
https://en.wikipedia.org/wiki/Rubber_duck_debugging

**Comment by corysimmons** [comment]
@tz5514 Seems like they're working on a PR for exactly what you're interested in: https://github.com/zeit/styled-jsx/pull/148

**Comment by giuseppeg** [comment]
fyi this should be possible when we implement/fix #169

**Comment by tz5514** [comment]
@corysimmons Looks great, if it will be implement in future plan, that's amazing
